src/main/java/cs/ualberta/cmput402/tictactoe/TicTacToeGame.java

```diff
@@ -12,44 +13,71 @@
 public class TicTacToeGame {
 
     private Board board;
+    private Scoreboard scoreboard;
+    private boolean gameFinished;
 
-    public TicTacToeGame(){
+    public TicTacToeGame() {
         board = new Board();
+        scoreboard = new Scoreboard();
+        gameFinished = false;
     }
 
-    public void promptNextPlayer(){
-        switch(board.getCurrentPlayer()){
+    public void promptNextPlayer() {
+        switch (board.getCurrentPlayer()) {
             case X:
                 System.out.println("It's player " + board.getSymbol(board.getCurrentPlayer()) + "'s turn. Please enter the coordinates of your next move as x,y: ");
                 break;
             case O:
                 System.out.println("It's player " + board.getSymbol(board.getCurrentPlayer()) + "'s turn. Please enter the coordinates of your next move as x,y: ");
                 break;
-
         }
     }
 
-    public void playGame(){
+    public void playGame() {
         Scanner keyboardScanner = new Scanner(System.in);
 
-        while (board.getWinner() == null){
+        while (!gameFinished) {
+            while (board.getWinner() == null && !board.getIsTie()) {
+                board.printBoard();
+                promptNextPlayer();
+                String line = keyboardScanner.nextLine();
+                String input[] = line.split(",");
+                try {
+                    board.playMove(Integer.parseInt(input[0]), Integer.parseInt(input[1]));
+                } catch (InvalidMoveException e) {
+                    System.out.println("Invalid coordinates. Try again");
+                    promptNextPlayer();
+                }
+            }
+
             board.printBoard();
-            promptNextPlayer();
+
+            if (board.getIsTie()) {
+                scoreboard.setTie(scoreboard.getTie() + 1);
+                System.out.println("It is a tie!");
+            } else {
```

Comment:
Could add some comments here, took me a second to understand the nested else.

Reply:
Just added some comments to the code.